usr/src/uts/common/fs/zfs/dmu_traverse.c

```diff
@@ -253,7 +254,7 @@ traverse_visitbp(traverse_data_t *td, const dnode_phys_t *dnp,
 		 *
 		 * Note that the meta-dnode cannot be reallocated.
 		 */
-		if ((!td->td_realloc_possible ||
+		if (!ignore_hole_birth && (!td->td_realloc_possible ||
```

Comment:
I think we should be a bit more explicit about the fact that holes with birth time are always accurate (right?); it's only holes without birth time that may have actually not existed since the beginning of time.  `ignore_hole_birth` implies that we are ignoring holes with birth time, which this doesn't do.  Perhaps something like `send_holes_without_birth_time` would be more accurate?